modular_skyrat/modules/primitive_fun/code/big_mortar.dm

```diff
@@ -0,0 +1,116 @@
+
+/obj/structure/large_mortar/examine(mob/user)
+	. = ..()
+	. += span_notice("It currently contains <b>[LAZYLEN(contents)]/[maximum_contained_items] items.")
+	. += span_notice("It can be (un)secured with <b>Right Click</b>")
+	. += span_notice("You can empty all of the items out of it with <b>Alt Click</b>")
+
+/obj/structure/large_mortar/Destroy()
+	drop_everything_contained()
+	return ..()
+
+/obj/structure/large_mortar/AltClick(mob/user)
+	if(!LAZYLEN(contents))
+		balloon_alert(user, "nothing inside")
+		return
+	drop_everything_contained()
+	balloon_alert(user, "removed all items")
+
+/// Drops all contents at the mortar
+/obj/structure/large_mortar/proc/drop_everything_contained()
+	if(!LAZYLEN(contents))
+		return
+	for(var/obj/target_item in contents)
+		target_item.forceMove(get_turf(src))
+		LAZYREMOVE(contents, target_item)
+
+/obj/structure/large_mortar/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(!can_interact(user) || !user.canUseTopic(src, be_close = TRUE))
+		return
+	set_anchored(!anchored)
+	balloon_alert_to_viewers(anchored ? "secured" : "unsecured")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/structure/large_mortar/attackby(obj/item/attacking_item, mob/living/carbon/human/user)
+	if(istype(attacking_item, /obj/item/pestle))
+		if(!anchored)
+			balloon_alert(user, "secure to ground first")
+			return
+		if(!LAZYLEN(contents))
+			balloon_alert(user, "nothing to grind")
+			return
+		if(user.getStaminaLoss() > 50)
+			balloon_alert(user, "too tired")
+			return
+		var/list/choose_options = list(
+			"Grind" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_grind"),
+			"Juice" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_juice")
+		)
+		var/picked_option = show_radial_menu(user, src, choose_options, radius = 38, require_near = TRUE)
+		if(!LAZYLEN(contents) || !in_range(src, user) || !user.is_holding(attacking_item) && !picked_option)
+			return
+		balloon_alert_to_viewers("grinding...")
+		if(!do_after(user, 5 SECONDS, target = src))
+			balloon_alert_to_viewers("stopped grinding")
+			return
+		user.adjustStaminaLoss(60) //This is a bit more tiring than a normal sized mortar and pestle
+		switch(picked_option)
+			if("Juice")
+				for(var/obj/item/target_item in contents)
+					if(target_item.juice_results)
+						juice_target_item(target_item, user)
+					else
+						grind_target_item(target_item, user)
+			if("Grind")
+				for(var/obj/item/target_item in contents)
+					if(target_item.grind_results)
+						grind_target_item(target_item, user)
+					else
+						juice_target_item(target_item, user)
+		return
+	if(LAZYLEN(contents) >= maximum_contained_items)
+		balloon_alert(user, "already full")
+		return
+	if(!attacking_item.juice_results || !attacking_item.grind_results)
+		balloon_alert(user, "can't grind this")
+		return
+	attacking_item.forceMove(src)
+	LAZYADD(contents, attacking_item)
```

Comment:
this logic feels misplaced/weird... I could imagine a non-grindable/juiceable item could sometimes get the "already full" warning. And since you also did the return ..() change, this logic really needs to be changed somehow.